packages/rule-no-broken-links/src/rule.ts

```diff
@@ -0,0 +1,181 @@
+            }
+
+            return null;
+        };
+
+        /**
+         * The callback to handle when an element is visited
+         * We will get the url(href,src etc) and check if it is available online
+         * We do not need to check the items received from fetch::end::* event
+         */
+        const validateElementSrcs = async (traverseElement: ElementFound): Promise<void> => {
+            const { element, resource } = traverseElement;
+            const simpleAttributes: Array<string> = ['src', 'poster', 'data', 'href'];
+
+            const urls: Array<string> = simpleAttributes.reduce((found: Array<string>, attribute: string) => {
+                const value: string = element.getAttribute(attribute);
+
+                if (value) {
+                    found.push(value);
+                }
+
+                return found;
+            }, []);
+
+            const srcset: Array<string> = parseSrcSet(element.getAttribute('srcset'));
+
+            if (srcset.length > 0) {
+                urls.push(...srcset);
+            }
+
+            const reports: Array<Promise<void>> = urls.map(async (url) => {
+                const fullUrl = URL.resolve(resource, url);
+
+                const fetched = getFetchedUrl(fullUrl);
+
+                if (fetched) {
+                    const statusIndex = brokenStatusCodes.indexOf(fetched.statusCode);
+
+                    if (statusIndex > -1) {
+                        return context.report(fullUrl, null, `Broken link found (${brokenStatusCodes[statusIndex]} response)`);
+                    }
+                } else {
+                    // An element which was not present in the fetch end results
+                    return await requester
+                        .get(fullUrl)
+                        .then((value: NetworkData) => {
+                            return handleSuccess(value, fullUrl, element);
+                        })
+                        .catch((error: any) => {
+                            return handleRejection(error, fullUrl, element);
+                        });
+                }
+
+                return Promise.resolve();
+            });
+
+            await Promise.all(reports);
+        };
+
+        /**
+         * Handler for fetch::end::* event.
+         * We will store the request url and response status code in fetchedUrls array
+         */
+        const validateFetchEnd = (fetchEnd: any) => {
+            fetchedUrls.push({ statusCode: fetchEnd.response.statusCode, url: fetchEnd.resource });
+        };
+
+        context.on('element::img', validateElementSrcs);
+        context.on('element::a', validateElementSrcs);
+        context.on('element::audio', validateElementSrcs);
+        context.on('element::video', validateElementSrcs);
+        context.on('element::source', validateElementSrcs);
+        context.on('element::track', validateElementSrcs);
+        context.on('element::object', validateElementSrcs);
```

Comment:
Should we add `link` and `script` to this list as well?

Reply:
I did add those 2